pkg/test/framework/components/echo/docker/instance.go

```diff
@@ -98,14 +98,12 @@ func newInstance(ctx resource.Context, cfg echo.Config) (out *instance, err erro
 	}()
 
 	// Validate the configuration.
-	if cfg.Galley == nil {
-		// Galley is not actually required currently, but it will be once Pilot gets
-		// all resources from Galley. Requiring now for forward-compatibility.
-		return nil, errors.New("galley must be provided")
-	}
 	if cfg.Pilot == nil {
 		return nil, errors.New("pilot must be provided")
 	}
+	if cfg.Cluster == nil {
```

Comment:
I don't think you need the `if` ... I think `ClusterOrDefault` does this for you.